proto/sf/ethereum/type/v2/type.proto

```diff
@@ -883,6 +883,11 @@ message BalanceChange {
   // The block's global ordinal when the balance change was recorded, refer to [Block]
   // documentation for further information about ordinals and total ordering.
   uint64 ordinal = 5;
+
+  // WithdrawalIndex links this balance change to a specific withdrawal when reason is REASON_WITHDRAWAL.
+  // This is the index in Block.withdrawals that caused this balance change.
+  // Only present for REASON_WITHDRAWAL balance changes.
+  optional uint64 withdrawal_index = 6;
```

Comment:
Why this change? Don't we have all the data in the new `repeat Withdrawal` already? Ideally we would avoid that, the fact that it's withdrawal only also seems to say that it should be on balance change.